src/UseCases/ValidateMembershipFee/ValidateMembershipFeeUseCase.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\MembershipContext\UseCases\ValidateMembershipFee;
+
+use InvalidArgumentException;
+use WMDE\Euro\Euro;
+
+/**
+ * @license GNU GPL v2+
+ */
+class ValidateMembershipFeeUseCase {
+
+	private const MIN_PERSON_YEARLY_PAYMENT_IN_EURO = 24;
+	private const MIN_COMPANY_YEARLY_PAYMENT_IN_EURO = 100;
+	private const MONTHS_PER_YEAR = 12;
+
+	public const APPLICANT_TYPE_COMPANY = 'firma';
+	public const APPLICANT_TYPE_PERSON = 'person';
+
+	private $membershipFee;
+	private $paymentIntervalInMonths;
+	private $applicantType;
+
+	public function validate( ValidateFeeRequest $request ): ValidateFeeResult {
+		$this->membershipFee = $request->getMembershipFee();
+		$this->paymentIntervalInMonths = $request->getPaymentIntervalInMonths();
+		$this->applicantType = $request->getApplicantType();
+
+		try {
+			$amount = Euro::newFromString( $this->membershipFee );
```

Comment:
Currently, this is inconsistent with the donation part, where the request already gets a `Euro` object. See https://phabricator.wikimedia.org/T204202

Seeing that our application might be used in different locales (WM CH is interested in it), we should decide how we want to handle "localized" amounts. IMHO, localization can be considered part of the "framework" layer, so the use cases don't have to bother with it. Do you agree?

Do you have any preference or argument why a "Request" value object for a use case should get a string vs a Euro object? The value will be pre-processed by the framework layer anyway.

Not saying you should change this here, only asking for your thoughts on the matter,

Reply:
Localization is in my understanding a presentation concern so should indeed be done in FF.

I very briefly thought about this while moving the code and considered that if an Euro is passed, you'd get an error for negative numbers in FF. Thinking about that more now I don't see why that would be a problem. So yeah, passing Euro sounds good.

Makes me wonder why the membership stuff uses int instead of Euro. Most of the membership stuff was written after the donation stuff. Perhaps we just forgetti'd it here or the donation stuff got refactored...

Reply:
Going to change this. When I wrote my previous comment I was under the incorrect impression that the current request uses and integer while it actually uses a string. That while you commented on a line with `Euro::newFromString`. Provides I am blind AF :D